File: src/main/java/com/rana/demologin/service/AuthService.java
package com.rana.demologin.service;

import com.rana.demologin.dto.LoginRequest;
import com.rana.demologin.dto.RegisterRequest;
import com.rana.demologin.model.User;
import com.rana.demologin.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthService {
    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtService jwtService;
    
    public String login(LoginRequest request) {
        User user = userRepository.findByUsername(request.getUsername())
            .orElseThrow(() -> new RuntimeException("User not found"));
            
        if (!passwordEncoder.matches(request.getPassword(), user.getPassword())) {
            throw new RuntimeException("Invalid password");
        }
        
        return jwtService.generateToken(user);
    }
    
    public void register(RegisterRequest request) {
        if (userRepository.existsByUsername(request.getUsername())) {
            throw new RuntimeException("Username already exists");
        }
        
        if (userRepository.existsByEmail(request.getEmail())) {
            throw new RuntimeException("Email already exists");
        }
        
        User user = new User();
        user.setUsername(request.getUsername());
        user.setPassword(passwordEncoder.encode(request.getPassword()));
        user.setEmail(request.getEmail());
        user.setFullName(request.getFullName());
        
        userRepository.save(user);
    }
} 

File: src/main/kotlin/com/rana/demologin/service/AuthService.kt
package com.rana.demologin.service

import com.rana.demologin.dto.LoginRequest
import com.rana.demologin.dto.RegisterRequest
import com.rana.demologin.model.User
import com.rana.demologin.repository.UserRepository
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

@Service
class AuthService(
    private val userRepository: UserRepository,
    private val passwordEncoder: PasswordEncoder,
    private val jwtService: JwtService
) {
    fun login(request: LoginRequest): String {
        val user = userRepository.findByUsername(request.username)
            ?: throw RuntimeException("User not found")
            
        if (!passwordEncoder.matches(request.password, user.password)) {
            throw RuntimeException("Invalid password")
        }
        
        return jwtService.generateToken(user)
    }
    
    fun register(request: RegisterRequest) {
        if (userRepository.existsByUsername(request.username)) {
            throw RuntimeException("Username already exists")
        }
        
        if (userRepository.existsByEmail(request.email)) {
            throw RuntimeException("Email already exists")
        }
        
        val user = User(
            username = request.username,
            password = passwordEncoder.encode(request.password),
            email = request.email,
            fullName = request.fullName
        )
        
        userRepository.save(user)
    }
} 